sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -299,6 +301,11 @@ case class TakeOrderedAndProjectExec(
     }
   }
 
+  private def prepareAndWaitForSubqueries(): Unit = {
```

Comment:
instead of adding this method, shall we just call `executeQuery` in `executeCollect`?

Reply:
e.g. https://github.com/apache/spark/pull/43419/files#r1363168594

Reply:
Sg. Just want to make sure `RDDOperationScope.withScope` does not any side effect right?

Reply:
It's the same with this one, with some tracking logic